tests/v2/integration/catalogv2/system_charts_test.go

```diff
@@ -0,0 +1,439 @@
+	w.Require().NoError(err)
+	w.Assert().Equal(app.Spec.Chart.Metadata.Version, latest)
+}
+
+func (w *SystemChartsTest) TestUpgradeChartToLatestVersion() {
+	defer w.resetSettings()
+
+	clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	origClusterRepo := clusterRepo.DeepCopy()
+	cfg, err := w.client.Steve.SteveType(configmaps.ConfigMapSteveType).ByID(fmt.Sprintf("%s/%s", origClusterRepo.Status.IndexConfigMapNamespace, origClusterRepo.Status.IndexConfigMapName))
+	w.Require().NoError(err)
+	cfgMap := &v1.ConfigMap{}
+	w.Require().NoError(stevev1.ConvertToK8sType(cfg.JSONResp, cfgMap))
+	origCfg := cfgMap.DeepCopy()
+
+	// GETTING INDEX FROM CONFIGMAP AND MODIFYING iT
+	gz, err := gzip.NewReader(bytes.NewBuffer(cfgMap.BinaryData["content"]))
+	w.Require().NoError(err)
+	defer gz.Close()
+	data, err := io.ReadAll(gz)
+	w.Require().NoError(err)
+	index := &repo.IndexFile{}
+	w.Require().NoError(json.Unmarshal(data, index))
+	index.SortEntries()
+	latestVersion := index.Entries["rancher-aks-operator"][0].Version
+	index.Entries["rancher-aks-operator"] = index.Entries["rancher-aks-operator"][1:]
+	//index.Entries["rancher-aks-operator-crd"] = index.Entries["rancher-aks-operator-crd"][1:]
+	marshal, err := json.Marshal(index)
+	w.Require().NoError(err)
+	var compressedData bytes.Buffer
+	writer := gzip.NewWriter(&compressedData)
+	_, err = writer.Write(marshal)
+	w.Require().NoError(err)
+	w.Require().NoError(writer.Close())
+	cfgMap.BinaryData["content"] = compressedData.Bytes()
+
+	//UPDATING THE CONFIGMAP
+	cfg, err = w.client.Steve.SteveType(configmaps.ConfigMapSteveType).Update(cfg, cfgMap)
+	w.Require().NoError(stevev1.ConvertToK8sType(cfg.JSONResp, cfgMap))
+
+	//Updating the cluster
+	w.Require().NoError(w.updateManagementCluster())
+
+	app, err := w.waitForAksChart(rv1.StatusDeployed, "rancher-aks-operator", 0)
+	w.Require().NoError(err)
+
+	latest, err := w.catalogClient.GetLatestChartVersion("rancher-aks-operator")
+	w.Require().NoError(err)
+	w.Assert().Equal(latest, app.Spec.Chart.Metadata.Version)
+	w.Assert().Greater(latestVersion, latest)
+
+	//REVERT CONFIGMAP TO ORIGINAL VALUE
+	cfgMap.BinaryData["content"] = origCfg.BinaryData["content"]
+	cfg, err = w.client.Steve.SteveType(configmaps.ConfigMapSteveType).Update(cfg, cfgMap)
+	w.Require().NoError(err)
+
+	clusterRepo, err = w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	clusterRepo.Spec.ForceUpdate = &metav1.Time{Time: time.Now()}
+	_, err = w.catalogClient.ClusterRepos().Update(context.TODO(), clusterRepo.DeepCopy(), metav1.UpdateOptions{})
+	w.Require().NoError(err)
+
+	app, err = w.waitForAksChart(rv1.StatusDeployed, "rancher-aks-operator", app.Spec.Version)
+	w.Require().NoError(err)
+
+	w.Assert().Equal(latestVersion, app.Spec.Chart.Metadata.Version)
+}
+
+func (w *SystemChartsTest) TestUpgradeToWorkingVersion() {
```

Comment:
Can we remove this testcase ? since we are already testing the case TestUpgradetoLatestVersion and latestversion is workingversion in this case?

Reply:
We need this as discussed before